### DISCUSSION TITLE
Как насчет того чтобы в будущих версиях упростить Uow

### DISCUSSION THREAD

**Discussion by Gankov** [QualitySolution/QSProjects#474] [Ideas]
Предлагаю оставить только одну реализацию Uow без наличия рутового объекта. Один обычный UnitOfWork.
Я все больше прихожу к выводу что такое деление на самом деле достаточно безсмысленное.
Оно ничего толком не улучшает, кроме как избавляет в коде от вызова Commit. Но так как при другом типе Uow все равно нужно вызывать этот самый Commit то у разработчиков это вызывает путаницу. Когда вызывать, когда нет.
Ну и разное поведение тоже вызывает путаницу и необходимость объяснять нюансы.

Ну и разные типы UnitOfWork еще и множат код в фабрике...

Кто что думает?

**Comment by Enzogord** [comment]
Согласен, лучше убрать его.

**Comment by Gankov** [comment]
Ок. В ближайшее время помечу все планируемое к удалению устаревшим.
Ну и в районе версии библиотек 1.10. Удалим.

**Comment by Gankov** [comment]
Кстати до меня тут дошло что такой рефакторинг позволит значительно упростить там работу с UnitOfWork, так как позволит предположить создание основного UnitOfWork на плечи AutoFac, и получать его почти везде просто в конструкторе. И не заботится о его дестрое. Надо делать. Будет сильно проще.
